src/main/java/webserver/HttpRequest.java

```diff
@@ -0,0 +1,153 @@
+    private static final Logger logger = LoggerFactory.getLogger(HttpRequest.class);
+
+    private static final String NOT_EXIST_MESSAGE = "(이)가 존재하지 않습니다.";
+
+    private final HttpMethod httpMethod;
+    private final String path;
+    private final Map<String, String> header;
+    private final Map<String, String> parameters;
+
+    private HttpRequest(HttpMethod httpMethod, String path, Map<String, String> header, Map<String, String> parameters) {
+        this.httpMethod = httpMethod;
+        this.path = path;
+        this.header = header;
+        this.parameters = parameters;
+    }
+
+    public static HttpRequest from(InputStream in) {
+        try {
+            BufferedReader br = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+            List<String> requestLine = readRequestLine(br);
+            HttpMethod httpMethod = getHttpMethodFromRequestLine(requestLine);
+            String path = getPathFromRequestLine(requestLine);
+            Map<String, String> parameters = getParametersFromRequestLine(requestLine);
+
+            List<String> requestHeader = readRequestHeader(br);
+            Map<String, String> header = getHeaderFromRequestHeader(requestHeader);
+
+            int contentLength = header.containsKey("Content-Length") ? Integer.parseInt(header.get("Content-Length")) : 0;
+            String requestBody = readRequestBody(br, contentLength);
+            parameters.putAll(getParametersFromRequestBody(requestBody));
+            return new HttpRequest(httpMethod, path, header, parameters);
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
+        }
+    }
+
+    private static List<String> readRequestLine(BufferedReader br) throws IOException {
+        return new ArrayList(Arrays.asList(br.readLine().split(" ")));
+    }
+
+    private static HttpMethod getHttpMethodFromRequestLine(List<String> requestLine) {
+        return HttpMethod.valueOf(requestLine.get(0));
+    }
+
+    private static String getPathFromRequestLine(List<String> requestLine) {
+        String path = requestLine.get(1);
+        int idx = path.indexOf('?');
+        if (idx != -1) {
+            path = path.substring(0, idx);
+        }
+        return path.equals("/") ? "/index.html" : path;
+    }
+
+    private static Map<String, String> getParametersFromRequestLine(List<String> requestLine) {
+        Map<String, String> parameters = new HashMap<>();
+        String line = requestLine.get(1);
+        int idx = line.indexOf('?');
+        if (idx != -1) {
+            String[] queries = line.substring(idx+1).split("&");
+            parameters = getParametersFromQueries(queries);
+        }
+        return parameters;
+    }
+
+    private static List<String> readRequestHeader(BufferedReader br) throws IOException {
+        List<String> header = new ArrayList<>();
+        String line = br.readLine();
+        while (line != null && !"".equals(line)) {
```

Comment:
이미 null check가 되어 있는 상태라서,
```suggestion
        while (line != null && !line.isEmpty()) {
```
해도 무방할것 같습니다. (isEmpty가 "".equals 보다 싸게 먹혀요)